### ISSUE TITLE
Resolve `<>`?

### ISSUE THREAD

**Issue by dblanovschi** [Mcat12/intellij-lalrpop#3]
As of now, (I think the Rust plugin?) says "error: Chained comparison operator require parentheses" when using `<>`

**Comment by Mcat12** [comment]
Yeah, this is the next big area of improvement for the plugin. A few things need to happen to support this notation:
1. We need to be able to modify the injected Rust action code such that `<>` is replaced with some special variable (e.g. `__intellij_lalrpop_selection`) so the Rust plugin can parse the code. The hard part is performing this transformation without affecting the source PSI elements (what's shown in the lalrpop file). This will require some more research.
2. Parse the "alternative" grammar so we know what the input variables and types are. Examples: `Item*` is a `Vec` of `Item`'s type, `<Visibility> "fn" <Name>` is a tuple of the visibility and name types, `<var:Id> "=" <expr:Expression>` has two named variables, etc. Currently, the last example is supported as long as `Id` and `Expression` explicitly declare their types, i.e. `Id: String = ...`

**Comment by dblanovschi** [comment]
I tried my hand at solving the second part and realized that in cases like `<Visibility> "fn" <Name>`, `<>` is not really a tuple, but `(<>)` is. Small but important distinction here.

**Comment by dblanovschi** [comment]
I might have just found a solution to this: the LiteralTextEscaper
```java
// from PsiLanguageInjectionHost
  /**
   * @return {@link LiteralTextEscaper} instance which will be used to convert the content of this host element to the content of injected file
   */
  @NotNull
  LiteralTextEscaper<? extends PsiLanguageInjectionHost> createLiteralTextEscaper();
```

**Comment by dblanovschi** [comment]
Ok turns out I was right: it is able to do what's needed to acheive that. Implemented it in https://github.com/dblanovschi/intellij-lalrpop/commit/745608f234e9bfa99d114738b879aa3401d4d3ef :tada: . Now on to evaluating what's "selected" from the list of symbols in the alternative.

**Comment by dblanovschi** [comment]
I finally got it to find what's selected or not (after getting sidetracked on type resolution in lalrpop; so now the types in rust injections are way more accurate).

**Comment by Mcat12** [comment]
Fixed by #17 :tada:
